extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -105,12 +108,29 @@
     };
 
     static final DotName OBJECT_MAPPER = DotName.createSimple("com.fasterxml.jackson.databind.ObjectMapper");
+    private static final Set<String> SASL_PROVIDERS = Arrays.stream(new String[] {
+            "com.sun.security.sasl.ClientFactoryImpl",
+            "com.sun.security.sasl.ServerFactoryImpl",
+            "com.sun.security.sasl.digest.FactoryImpl",
+            "com.sun.security.sasl.gsskerb.FactoryImpl",
+            "com.sun.security.sasl.ntlm.FactoryImpl",
+            "org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClient$OAuthBearerSaslClientFactory",
+            "org.apache.kafka.common.security.plain.internals.PlainSaslServer$PlainSaslServerFactory",
+            "org.apache.kafka.common.security.scram.internals.ScramSaslClient$ScramSaslClientFactory"
```

Comment:
I think what we need instead is something like:
```
"com.sun.security.sasl.Provider",
"org.apache.kafka.common.security.scram.internals.ScramSaslClientProvider",
"org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientProvider"
```

Note that we need to provide Providers through `-H:AdditionalSecurityProviders` or Service Types providing a "getInstance" method through `-H:AdditionalSecurityServiceTypes`, per https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/JCASecurityServices.md

Reply:
I have pushed this change to https://github.com/zakkak/quarkus/tree/native_additional_security_provider and started a new CI run to test it https://github.com/graalvm/mandrel/actions/runs/962342308

Reply:
@sberyozkin it works https://github.com/graalvm/mandrel/runs/2889846463?check_suite_focus=true#step:9:467

Reply:
Hi @zakkak, welcome back, thanks for looking into it, yes I was thinking adding those providers in a follow up step, but not instead of those factory classes but in addition to them...
So, re `-H:AdditionalSecurityServiceTypes`, perhaps we should register those factories with this option, in addition to registering the related 3 providers with `-H:AdditionalSecurityServiceProviders` ? Can you please add them manually and if it helps then I can add another build item.

Reply:
@zakkak Please ignore my last comment, it is a Liquibase test which is failing in your branch - this PR from @geoand may have fixed it: #18059. Can you please rebase ? Or I can apply your changes directly to this PR and rebase, let me know please if it would be easier

Reply:
Ack, rebased and started https://github.com/graalvm/mandrel/actions/runs/963808610

Reply:
@sberyozkin all tests are green!

Reply:
@zakkak Super great, thanks :-)

Reply:
@zakkak Can you please push your update to this PR's branch ?

Reply:
Nope, it doesn't let me do so.

```
Pushing to github.com:sberyozkin/quarkus.git
ERROR: Permission to sberyozkin/quarkus.git denied to zakkak.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Reply:
@zakkak Oh sorry, I guess a PR to my branch would be best then, I've also sent an invite

Reply:
Going to quickly update my PR as agreed :-)

Reply:
Done, will merge once the PR build passes

Reply:
I started https://github.com/graalvm/mandrel/actions/runs/964684774 as well to be on the safe side :)